src/hotspot/share/gc/g1/g1BlockOffsetTable.hpp

```diff
@@ -128,7 +124,9 @@ class G1BlockOffsetTable: public CHeapObj<mtGC> {
     }
   }
 
-  void set_for_starts_humongous(HeapRegion* hr, HeapWord* obj_top, size_t fill_size);
+  void update_for_block(HeapWord* blk_start, size_t size) {
```

Comment:
Maybe this can be done at the caller side? (The API would be easier with fewer overloading.)

Reply:
I removed the method `void update_for_block(HeapWord* blk_start, size_t size)` just now.